pilot/pkg/xds/filters/filters.go

```diff
@@ -163,6 +163,37 @@ var (
 	}
 
 	HTTPMx = buildHTTPMxFilter()
+
+	CaptureTLS = &listener.Filter{
+		Name: "capture_tls",
+		ConfigType: &listener.Filter_TypedConfig{
+			TypedConfig: protoconv.TypedStruct("type.googleapis.com/istio.tls_passthrough.v1.CaptureTLS"),
+		},
+	}
+
+	RestoreTLS = &listener.Filter{
+		Name: "restore_tls",
+		ConfigType: &listener.Filter_TypedConfig{
+			TypedConfig: protoconv.TypedStruct("type.googleapis.com/istio.tls_passthrough.v1.RestoreTLS"),
+		},
+	}
+
+	Baggage = &hcm.HttpFilter{
+		Name: "istio.filters.http.baggage_handler",
+		ConfigType: &hcm.HttpFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStruct("type.googleapis.com/istio.telemetry.baggagehandler.v1.Config"),
+		},
+	}
+	SetDstAddress = &listener.ListenerFilter{
+		Name: "set_dst_address",
+		ConfigType: &listener.ListenerFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStruct("type.googleapis.com/istio.set_internal_dst_address.v1.Config"),
+		},
+	}
+	MetadataToPeerNode = &listener.ListenerFilter{
+		Name:       "envoy.filters.listener.metadata_to_peer_node",
```

Comment:
I know these are from ambient mesh, where can I find the docs about it

Reply:
* set_dst_address is for internal listeners to grab ORIGINAL_DST from endpoint metada. Example endpoint points to "envoy://outbound_tunnel" internal listener, has endpoint metadata "destination: 1.2.3.4:15008", an orig dst cluster can forward to that. Allows us to let LB decide to do additional filters by going through an internal listener to reach some specific endpoint. 

* baggage_handler parses a header to use for telemetry

Reply:
Would be good to see a design doc for telemetry in hbone - the old model of passing a lot of redundant and unused data is not ideal. We also have the metadata in the workload cert - namespace, service account of the caller.

Reply:
I was also wondering the other day what/where captureTLS/restoreTLS are used for?